assets/shared/helpers/player/use-editor-player.js

```diff
@@ -133,8 +133,17 @@ const useEditorPlayer = ( videoBlock ) => {
 		const playerIframe = doc?.querySelector( 'iframe' );
 		const playerScript = doc?.getElementById( API_SCRIPT_ID );
 
-		// Skip if iframe is not found or player was already added.
-		if ( ! playerIframe || playerScript ) {
+		// Skip if iframe is not found.
+		if ( ! playerIframe ) {
+			return;
+		}
+
+		// If player script was already added, add load event to make sure player will be set.
+		if ( playerScript ) {
+			playerScript.addEventListener( 'load', () => {
```

Comment:
If we create 2 instances of the player at the same time, it wouldn't resolve for one because only one of them would listen to the `load` event.